src/Adapter/Discount/CommandHandler/UpdateDiscountConditionsHandler.php

```diff
@@ -51,6 +51,7 @@ public function handle(UpdateDiscountConditionsCommand $command): void
             $command->getMinimumAmountShippingIncluded(),
             $command->getCarrierIds() ? array_map(fn (CarrierId $carrierId) => $carrierId->getValue(), $command->getCarrierIds()) : null,
             $command->getCountryIds() ? array_map(fn (CountryId $countryId) => $countryId->getValue(), $command->getCountryIds()) : null,
+            $command->getCustomerGroupIds(),
```

Comment:
You should mimic what was done on other arrays of IDs:
- the command expect an array of integers
- the command transforms the integers into an array of value objects (in this case `CustomerGroupId`)
- the handler here transforms them back into an array of integers
 
This prevents strong coupling to a VO in the `DiscountConditionsUpdater` service, but it still gives internal validation on more consistent data via the CQRS command (and it follows our usual convention)